pkg/controller/bypass/bypass_test.go

```diff
@@ -0,0 +1,287 @@
+package bypass
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/agiledragon/gomonkey/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"kmesh.net/kmesh/pkg/utils"
+)
+
+func TestStartByPassKmesh(t *testing.T) {
+	client := fake.NewSimpleClientset()
+
+	err := os.Setenv("NODE_NAME", "test-node")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		os.Unsetenv("NODE_NAME")
+	})
+
+	addFuncCallCount := 0
+	delFuncCallCount := 0
+
+	patches1 := gomonkey.NewPatches()
+	patches2 := gomonkey.NewPatches()
+	defer patches1.Reset()
+	defer patches2.Reset()
+
+	podWithLabel := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod-with-label",
+			Namespace: "default",
+			Labels: map[string]string{
+				"kmesh.net/bypass": "enabled",
+			},
+		},
+		Spec: corev1.PodSpec{
+			NodeName: "test-node",
+		},
+	}
+
+	patches2.ApplyFunc(isKmeshManaged, func(pod *corev1.Pod) bool {
```

Comment:
nit we donot need to replace `isKmeshManaged`, which is simple checking pod annotations